x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -98,81 +96,111 @@ public record UnresolvedPolicy(String name, Enrich.Mode mode) {
-                Tuple<ResolvedEnrichPolicy, String> resolved = mergeLookupResults(
+                MergedPolicyLookupResult resolved = mergeLookupResults(
                     unresolved,
-                    calculateTargetClusters(unresolved.mode, includeLocal, remoteClusters),
+                    calculateTargetClusters(unresolved.mode, includeLocal, remotes),
                     lookupResponsesToProcess
                 );
 
-                if (resolved.v1() != null) {
-                    enrichResolution.addResolvedPolicy(unresolved.name, unresolved.mode, resolved.v1());
-                } else {
-                    assert resolved.v2() != null;
-                    enrichResolution.addError(unresolved.name, unresolved.mode, resolved.v2());
+                for (Map.Entry<String, Exception> entry : resolved.remotesToBeSkipped().entrySet()) {
+                    enrichResolution.addUnusableRemote(entry.getKey(), entry.getValue());
+                }
+                // If a remote-only CCS is done and all the remotes are "unusable" (due to missing enrich policies)
+                // *and* skippable (skip_unavailable=true), don't fill in either the resolved policy or the error on
+                // the enrichResolution object. The only field with contents will be the "unusable remotes" field.
+                if (resolved.resolvedPolicy() != null) {
+                    enrichResolution.addResolvedPolicy(unresolved.name, unresolved.mode, resolved.resolvedPolicy());
+                } else if (resolved.error() != null) {
+                    enrichResolution.addError(unresolved.name, unresolved.mode, resolved.error());
                 }
             }
             return enrichResolution;
         }));
     }
 
-    private Collection<String> calculateTargetClusters(Enrich.Mode mode, boolean includeLocal, Set<String> remoteClusters) {
+    record MergedPolicyLookupResult(
+        @Nullable ResolvedEnrichPolicy resolvedPolicy,
+        Map<String, Exception> remotesToBeSkipped,
+        @Nullable String error
+    ) {}
+
+    /**
+     * @param mode
+     * @param includeLocal
+     * @param remoteClusters
+     * @return Map where key: String=clusterAlias, value: Boolean=skipUnavailable setting
+     */
+    private Map<String, Boolean> calculateTargetClusters(Enrich.Mode mode, boolean includeLocal, Map<String, Boolean> remoteClusters) {
         return switch (mode) {
-            case ANY -> CollectionUtils.appendToCopy(remoteClusters, RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-            case COORDINATOR -> List.of(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
-            case REMOTE -> includeLocal
-                ? CollectionUtils.appendToCopy(remoteClusters, RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)
-                : remoteClusters;
+            case ANY -> copyAndAddLocalCluster(remoteClusters);
+            case COORDINATOR -> copyAndAddLocalCluster(Collections.emptyMap());
```

Comment:
Can't we just make it Map.of(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY, false) here?

Reply:
Yes,. Changed in next push.

Reply:
Good idea. Fixed in next push.